src/Fable.React/Fable.Helpers.React.fs

```diff
@@ -903,45 +905,82 @@ module ReactElementType =
+    /// classes.
+    ///
+    /// If your function component renders the same result given the same props, you can wrap it in a call to React.memo
+    /// for a performance boost in some cases by memoizing the result. This means that React will skip rendering the
+    /// component, and reuse the last rendered result.
+    ///
+    /// By default it will only shallowly compare complex objects in the props object. If you want control over the
+    /// comparison, you can use `memoWith`.
+    [<Import("memo", from="react")>]
+    let memo<'props> (render: 'props -> ReactElement) : ReactComponentType<'props> =
+        jsNative
+
+    [<Import("memo", from="react")>]
+    let private reactMemoWith<'props> (render: 'props -> ReactElement, areEqual: PropsEqualityComparison<'props>) : ReactComponentType<'props> =
+        jsNative
+
+    /// React.memo is a higher order component. It’s similar to React.PureComponent but for function components instead of
+    /// classes.
+    ///
+    /// If your function component renders the same result given the same props, you can wrap it in a call to React.memo
+    /// for a performance boost in some cases by memoizing the result. This means that React will skip rendering the
+    /// component, and reuse the last rendered result.
+    ///
+    /// This version allow you to control the comparison used instead of the default shallow one by provide a custom
+    /// comparison function.
+    let memoWith<'props> (areEqual: PropsEqualityComparison<'props>) (render: 'props -> ReactElement) : ReactComponentType<'props> =
+        reactMemoWith(render, areEqual)
+
+/// memo is similar to React.PureComponent but is built from only a render function.
 ///
-/// If your function component renders the same result given the same props, you can wrap it in a call to React.memo
-/// for a performance boost in some cases by memoizing the result. This means that React will skip rendering the
-/// component, and reuse the last rendered result.
+///
+/// If your function renders the same result given the same props, you can wrap it in a call to memo for a performance
+/// boost in some cases by memoizing the result. This means that React will skip rendering the component, and reuse the
+/// last rendered result.
+///
+/// The resulting function shouldn't be used directly in a render but should be stored to be reused :
+///
+/// ```
+/// type HelloProps = { Name: string }
+/// let hello = memo "Hello" (fun { Name = name } -> span [str "Hello "; str name])
+///
+/// let view model =
+///     hello { Name = model.Name }
+/// ```
 ///
 /// By default it will only shallowly compare complex objects in the props object. If you want control over the
 /// comparison, you can use `memoWith`.
-let memo<'props> (name: string) (render: 'props -> ReactElement) : ReactComponentType<'props> =
+let memo<'props> (name: string) (render: 'props -> ReactElement) : 'props -> ReactElement =
```

Comment:
Should we use the name `memoBuilder` here? This make it more obvious that the function is not a direct binding of `memoBuilder`, then we keep the `memo` direct binding in `ReactElementType` module for people who need it.

Reply:
I think the comment with sample is  enough to defuse the confusion